js/coinone.js

```diff
@@ -371,78 +371,253 @@ module.exports = class coinone extends Exchange {
-            if (price !== undefined) {
-                cost = price * amount;
-            }
-        }
-        const currency = this.safeString (info, 'currency');
-        const fee = {
-            'currency': currency,
-            'cost': this.safeFloat (info, 'fee'),
-            'rate': this.safeFloat (info, 'feeRate'),
-        };
-        let symbol = undefined;
-        if (market === undefined) {
-            const marketId = currency.toLowerCase ();
-            if (marketId in this.markets_by_id) {
-                market = this.markets_by_id[marketId];
+        let status = undefined;
+        const orderStatus = this.safeString (order, 'status');
+        if (orderStatus === 'live') {
+            status = 'open';
+            filled = 0;
+            if (amount > remaining) {
+                amount = remaining;
+                remaining = 0;
             }
+        } else if (orderStatus === 'partially_filled') {
+            status = 'open';
+            filled = amount - remaining;
+        } else {    // 'filled'
+            status = 'closed';
+            filled = amount - remaining;
         }
-        if (market !== undefined) {
-            symbol = market['symbol'];
-        }
+        const cost = price * filled;
         return {
-            'info': order,
             'id': id,
             'clientOrderId': undefined,
-            'timestamp': timestamp,
             'datetime': this.iso8601 (timestamp),
+            'timestamp': timestamp,
             'lastTradeTimestamp': undefined,
+            'status': status,
             'symbol': symbol,
             'type': 'limit',
             'side': side,
             'price': price,
-            'cost': cost,
             'amount': amount,
             'filled': filled,
             'remaining': remaining,
-            'status': status,
-            'fee': fee,
-            'average': undefined,
+            'cost': cost,
             'trades': undefined,
+            'fee': {
+                'currency': feeCurrency,
+                'cost': this.safeFloat (info, 'fee'),
+                'rate': this.safeFloat (info, 'feeRate'),
+            },
+            'info': order,
         };
     }
 
+    async fetchOpenOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        // [WARNING] Because of the limited capabilities of Coinone V2 API, you should be noticed below
+        // * The symbol parameter must be given. If not, an exception will occur.
+        // * Always returns the list of the all open orders with the symbol: The since and limit parameters are totally ignored.
+        // * The returned amount might not be same as the ordered amount. If an order is partially filled, the returned amount means the remaining amount.
+        // * For the same reason, the returned amount and remaining are always same, and the returned filled and cost are always zero.
+        await this.loadMarkets ();
+        if (symbol === undefined) {
```

Comment:
I would move this check above `await this.loadMarkets()` to throw as soon as possible, before even attempting the network request for loading the markets.

Reply:
Agreed